src/components/common/Dialogs/ControlSafeDialog/types.ts

```diff
@@ -17,6 +18,26 @@ export interface ControlSafeProps {
   // setStepIndex: React.Dispatch<React.SetStateAction<number>>;
 }
 
+export interface FunctionParamType {
+  name: string;
+  type: string;
+}
+
+export type SafeTransaction = {
+  transactionType: string;
+  token?: Token;
+  amount?: string;
+  rawAmount?: number;
+  recipient?: User;
+  data: string;
+  contract?: SimpleTarget;
+  abi: string;
+  contractFunction: string;
+  functionParamTypes?: FunctionParamType[];
+  nft?: SelectedPickerItem;
+  nftData?: NFTData;
+};
+
```

Comment:
I'm sorry but can you explain what you mean a bit further? Are you saying that we should use the `SafeTransaction` fragments that we have in `~types` or that we should move these types there?

Reply:
yes sorry for the confusion I am saying that these should be removed and that we should use the fragments that we have in `~types`

ie:
```
export type SafeTransactionData = SafeTransactionDataFragment;

export type NFTData = NftDataFragment;

export type FunctionParam = FunctionParamFragment;
```

in `src/types/graphql.ts`

Reply:
Ah I see, thank you for explaining. It may seem like it but the data structure and types of both objects have their differences, mostly because of validation purposes.

For example, `recipient`, `token`, and `rawAmount` have different types. On another note, `functionParamTypes` doesn't exist in the Fragment, it was renamed to `functionParams` and now it has 1 more prop called `value` inside it which is not the case for `functionParamTypes` (I tried to make it so but the validation was problematic.). 

We can each a compromise anyways by extending the `SafeTransactionData` type and omitting the props that don't match.